src/stores/FrameStore.ts

```diff
@@ -753,15 +727,33 @@ export class FrameStore {
         };
         this.animationChannelRange = [0, frameInfo.fileInfoExtended.depth - 1];
 
-        this.initSkyWCS();
-        if (frameInfo.fileInfoExtended.depth > 1) {
-            this.initFullWCS();
-        }
-
+        // init WCS
         this.astFrameSet = this.initFrame();
         if (this.astFrameSet) {
-            this.spectralFrame = AST.getSpectralFrame(this.astFrameSet);
+            if (this.spectralAxis && this.spectralAxis.valid) {
+                this.spectralFrame = AST.getSpectralFrame(this.astFrameSet);
+            }
+
+            if (frameInfo.fileInfoExtended.depth > 1) { // 3D frame
```

Comment:
I wonder... does this still work with 3D cubes with a single channel? (i.e. `NAXIS = 3` and `NAXIS3 = 1`)

Reply:
frameInfo.fileInfoExtended.depth > 1 is an existing old logic and spreading in quite some places, if it is incorrect, we can correct it in this PR

Reply:
`frameInfo.fileInfoExtended.depth > 1` is generally correct for checking whether a cube is 2D or 3D, but when it comes to AST, we might still need to check things more carefully.

Reply:
the original logic of 3D cube initialization is:
`if frameInfo.fileInfoExtended.depth > 1 then initFullWCS()`
and initFullWCS() gets NAXIS1~3 for AST

and current logic is just the same except adding a check for the spectral/stokes axis, so not sure the difference & have no clue about the checking,

@veggiesaurus would u mind elaborate that directly? thanks

Reply:
sorry, I think you're right, it shouldn't make a difference. I was just worried that situations where you've got an image with NAXIS = 3 and NAXIS3 = 1 (which for CARTA's purposes is really a 2D image, just with a degenerate 3rd axis), AST wouldn't let you use `astTran2` since the frame info would be 3D.